### PR TITLE
Use BaseAction class for RetrievalAction

### PR DESCRIPTION
**Pull request by PopDaph**
## Description

Following https://github.com/dust-tt/dust/pull/5043/files
Use our new BaseAction class for Retrieval action.

## Risk

Breaks retrieval action.
Can be rolled back.

## Deploy Plan

✅ Deploy Front Edge, check that both single & multi action agents work with retrieval.
Deploy Front.

### CODE REVIEW

File: front/lib/api/assistant/actions/retrieval.ts [hunk 1]
```diff
@@ -109,104 +113,130 @@ export function timeFrameFromNow(timeFrame: TimeFrame): number {
...
- * Model rendering of retrievals.
- */
+interface RetrievalActionBlob {
+  id: ModelId; // AgentRetrievalAction.
+  agentMessageId: ModelId;
+  params: {
+    relativeTimeFrame: TimeFrame | null;
+    query: string | null;
+    topK: number;
+  };
```
Comment (flvndvd) [inline front/lib/api/assistant/actions/retrieval.ts:123]: `nit` Let's extract this into a type.

**Review by flvndvd — APPROVED** [review]

File: types/src/front/assistant/actions/retrieval.ts [hunk 2]
```diff
@@ -4,6 +4,7 @@
 
 import { ModelId } from "../../../shared/model_id";
 import { ioTsEnum } from "../../../shared/utils/iots_utils";
+import { BaseAction } from "../../lib/api/assistant/actions";
```
Comment (spolu) [inline types/src/front/assistant/actions/retrieval.ts:7]: Must come from the root I beleive? so `../../../front`

Comment (spolu) [reply types/src/front/assistant/actions/retrieval.ts:7]: cc @lasryaric 

Also is it cool to use index files? This file really is `actions/index.ts`.

Comment (PopDaph) [reply types/src/front/assistant/actions/retrieval.ts:7]: https://github.com/dust-tt/dust/pull/5110

Comment (lasryaric) [reply types/src/front/assistant/actions/retrieval.ts:7]: Not sure why I am mentioned but yes we should `../../` up to root, and then go back down the tree.

**Pull request merged**